exchanges/sharedtestvalues/sharedtestvalues.go

```diff
@@ -47,3 +55,52 @@ func NewTestWebsocket() *stream.Websocket {
 		Match:             stream.NewMatch(),
 	}
 }
+
+// SkipTestIfCredentialsUnset is a test helper function checking if the
+// authenticated function can perform the required test.
+func SkipTestIfCredentialsUnset(t *testing.T, exch exchange.IBotExchange, canManipulateOrders ...bool) {
+	t.Helper()
+
+	if len(canManipulateOrders) > 1 {
+		t.Fatal("more than one canManipulateOrders boolean value has been supplied, please remove")
+	}
+
+	areTestAPICredentialsSet := AreAPICredentialsSet(exch)
+	supportsManipulatingOrders := len(canManipulateOrders) > 0
+	allowedToManipulateOrders := supportsManipulatingOrders && canManipulateOrders[0]
+
+	if areTestAPICredentialsSet && !supportsManipulatingOrders ||
+		areTestAPICredentialsSet && allowedToManipulateOrders {
+		return
+	}
+
+	message := []string{warningSkip}
+	if !areTestAPICredentialsSet {
+		message = append(message, warningKeys)
+	}
+
+	if supportsManipulatingOrders && !allowedToManipulateOrders {
+		message = append(message, warningManipulateOrders)
+	}
+	message = append(message, warningHowTo)
+	t.Skip(strings.Join(message, ", "))
+}
+
+// SkipTestIfCannotManipulateOrders will only skip if the credentials are set
+// correctly and can manipulate orders is set to false. It will continue normal
+// operations if credentials are not set, giving better code coverage.
+func SkipTestIfCannotManipulateOrders(t *testing.T, exch exchange.IBotExchange, canManipulateOrders bool) {
+	t.Helper()
+
+	if !AreAPICredentialsSet(exch) || canManipulateOrders {
+		return
+	}
+
+	message := []string{warningSkip, warningManipulateOrders}
+	t.Skip(strings.Join(message, ", "))
```

Comment:
```suggestion
t.Skip(warningSkip +", " + warningManipulateOrders)
```

https://www.youtube.com/watch?v=24ADMp-vrLg